common/src/test/org/apache/hadoop/hive/conf/TestHiveConf.java

```diff
@@ -95,6 +95,9 @@ public void testConfProperties() throws Exception {
 
     // Test HiveConf property variable substitution in hive-site.xml
     checkHiveConf("test.var.hiveconf.property", ConfVars.DEFAULTPARTITIONNAME.getDefaultValue());
+
+    // Test if LLAP_LRFU_HOTBUFFERS_PERCENTAGE is defined in HiveConf
+    Assert.assertTrue(HiveConf.getLlapDaemonConfVars().contains(ConfVars.LLAP_LRFU_HOTBUFFERS_PERCENTAGE.varname));
```

Comment:
wouldn't it be better to assert all llap daemon conf vars in a loop?

Reply:
Yeah! Okay good idea to test all conf vars at once.

Reply:
@abstractdog but only few confVars are part of llap daemon conf vars, so I will create a separate list of those and test them in loop

Reply:
I see, my point is that when some LLAP cache related patches were backported to downstream hive versions, there was a NoSuchFieldError and I'm wondering if we can make a future-proof test...like: if something is added to a collection, it should be added to / defined somewhere else

Reply:
Name starts with "LLAP_" would also not work because there are some conf vars which do not fall into this category.
I see your point but for now I am not sure if we could do anything on this to make this future proof.
Like for example there is a conf "LLAP_TASK_SCHEDULER_AM_REGISTRY_KEYTAB_FILE" which is not considered in llapDaemonVarsSetLocal

Reply:
if there are some options that are not considered to be present in llapDaemonVarsSetLocal, then we should add them as exclusions in the unit test, this way we can be future-proof:
1. dev adds a new "LLAP_" config
2. UT fails because it's not defined in llapDaemonVarsSetLocal
3. the dev has to think and add it to llapDaemonVarsSetLocal OR to the exclusion list in the unit test

Reply:
I see, this is good idea, I will added these changes, thanks.